vmware/build-tools-for-vmware-aria
# Store the form data as JSON rather than as serialized JSON into string.

### Description

When Custom Forms are pulled from Aria Automation, they are stored on the file system (the repo) in a form similar to 
```
	 {
	    "id": "e694a748-7067-47d1-91a4-614da73dda03",
	    "name": "Test",
	    "form": "{\"layout\": {...},\"schema\": {...},\"options\": {...}}",
	    "styles": null,
	    "sourceType": "com.vmw.blueprint",
	    "sourceId": "71ac6ebc-6a94-3c5a-8c00-2a44ddf81bce",
	    "type": "requestForm",
	    "status": "ON",
	    "formFormat": "JSON"
	 }
```
Here, please note that the form field is a double serialized JSON as string : "{\"layout\": {...},\"schema\": {...},\"options\": {...}}"
which makes it goes in one line and very difficult for a human to work with. If there is any commits and. changes in the form in the repo, the diffs are very difficult to find (when reviewing pull requests). 
As a whole it is not human friendly and very difficult for a human to deal with. 

The current pull request, makes it so that the format will become a properly formatted JSON object like: 
```
{
	    "id": "e694a748-7067-47d1-91a4-614da73dda03",
	    "name": "Test",
	    "form": {
	      "layout": {
	      ...
	      },
	      "schema": {
	      ...
	      },
	      "options": {
	      ...
	      }
	    },
	    "styles": null,
	    "sourceType": "com.vmw.blueprint",
	    "sourceId": "71ac6ebc-6a94-3c5a-8c00-2a44ddf81bce",
	    "type": "requestForm",
	    "status": "ON",
	    "formFormat": "JSON"
}
```
This way it is more easy to work with. 
### Checklist


- [X] I have added relevant error handling and logging messages to help troubleshooting
- [x] I have updated CHANGELOG.md with a short summary of the changes introduced
- [x] I have synced any structure and/or content vRA-NG improvements with vra-ng and ts-vra-ng archetypes (if applicable) - 
- [X] I have added a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue - NOT APPLICABLE

### Testing

On Actual Aria Automation installation (on the lab)
- Pull content to some VRA-NG Project that contains Catalog ItemsCustom Forms.
- Check that the custom forms metadata file in folder `catalog-item/forms` contain two JSON files. 
  * one with extension `-FormData.json` that contains just the Form Data in JSON format - formated.
  * another one without the `-FormData` suffix. This one contains all data about the custom form including metadata, but also the form itself in JSON format. The form should exist as a properly formatter JSON object and not as a string as it was before.
- Once the content is pulled and verified, push it back to make sure it is pushable.

### Release Notes

Export (Pull) Arai Automation Custom forms in a structured JSON, rather than in string encoded JSON, when stored together with the Custom Form metadata, thus making it more human friendly to operate with when stored in a repo.

### Related issues and PRs
